src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/SoftRateLimit.swift

```diff
@@ -0,0 +1,85 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import ExposureNotification
+
+
+class SoftRateLimit {
+	// MARK: - Init
+
+	init(
+		store: Store,
+		logger: Logging? = nil
+	) {
+		self.store = store
+		self.logger = logger
+	}
+
+	func setup(configuration: RiskProvidingConfiguration) -> Bool {
+		let enoughTimeHasPassed = configuration.shouldPerformExposureDetection(
+			lastExposureDetectionDate: store.exposureDetectionDate, context: " for soft rate limit"
+		)
+		let blocking = !enoughTimeHasPassed
+		if blocking {
+			Log.info("Soft rate limit is stricter than effective rate limit", log: .riskDetection, logger: logger)
+		} else {
+			Log.debug("Soft rate limit is in synch with effective rate limit", log: .riskDetection, logger: logger)
+		}
+		return blocking
+	}
+
+	func assess(
```

Comment:
Because we want to log something here, i suggest to name the method more like `logRateLimit` or something similar